src/components/WalletModal/index.tsx

```diff
@@ -156,15 +156,23 @@ export default function WalletModal({
   // When new wallet is successfully set by the user, trigger logging of Amplitude analytics event.
   useEffect(() => {
     if (account && account !== lastActiveWalletAddress) {
+      const walletType = getConnectionName(getConnection(connector).type, getIsMetaMask())
```

Comment:
in general i think when there's a decent amount of analytics code it's nice to split it out into a separate function to avoid mixing with UI code, but it's more of a judgement thing

Reply:
sounds good, moved amplitude stuff to other function sendAnalyticsEventAndUserInfo